skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -222,14 +247,16 @@ private Disposable addDownloadEventSubscription() {
                             break;
                         case FINISHED:
                             progressDialog.dismiss();
-                            Toast.makeText(this, getString(R.string.tv_form_send_success), Toast.LENGTH_SHORT).show();
+                            String result = downloadEvent.getResult();
+                            resultDialog.setMessage(result);
+                            resultDialog.show();
                             break;
                         case ERROR:
-                            progressDialog.dismiss();
+                            dismissAllDialogs();
```

Comment:
Ideally whichever dialog box is shown for showing the progress of form transfer, that should be dismissed here. Adding check for all the dialogs don't think its a good practice. Any reason for doing this way?

Reply:
I think that will  be much easier? 😢  All right, I'll fix this soon

Reply:
I'm not saying that we can't do this, but if we go this way then we may end up in some situation where two dialogs are shown at a single point of time.
That's why handle the show and dismiss of dialogs where it is expected to be done rather than closing all at once.

Reply:
I see, will improve.

Reply:
@lakshyagupta21 I keep this `dismissAll()` for `onStop()` method, and for `ERROR` case and `CANCEL` case, all we need to handle (dismiss) is progress dialog. Correct me if I'm wrong.

Reply:
`onStop `will be called when the app goes in the background right? So what if the user goes in background and comes in the foreground, this will dismiss all the dialog box no matter what operation is happening right? That's the reason we need to handle it carefully, otherwise unexpected situations can arise.

Reply:
When I write this code, I referred to `HpReceiverActivity.java`, that includes something like

```java
    @Override
    protected void onStop() {
        super.onStop();

        if (alertDialog != null && alertDialog.isShowing()) {
            alertDialog.dismiss();
        }
    }
```

I think we can just remove the `onStop()`, I'm wondering why we need to handle dialogs in `OnStop()`? @lakshyagupta21 right?

Reply:
I don't remember completely why was it added there, thanks for pointing it out we've to do some tests for activity transitions for hotspot as well.

Reply:
I will remove the `onStop()` and the `dismissAll` method here since I think we have already handle dialogs in `addDownloadEventSubscription()`, and we can check if it is also necessary in hotspot receiver and improve that.

Reply:
+1

Reply:
improved.